hack/preview.sh

```diff
@@ -83,6 +73,10 @@ if [ -n "$DOCKER_IO_AUTH" ]; then
     rm $AUTH
 fi
 
+rekor_server="rekor.$domain"
+sed -i "s/rekor-server.enterprise-contract-service.svc/$rekor_server/" $ROOT/argo-cd-apps/base/enterprise-contract.yaml
+yq -i e ".data |= .\"transparency.url\"=\"https://$rekor_server\"" $ROOT/components/build/tekton-chains/chains-config.yaml
```

Comment:
IIRC we don't set this currently, so we're using the public Rekor instance at rekor.sigstore.dev by default and we need to switch over manually by running `hack/chains/config.sh local-rekor` with Argo synchronization off for the Build application. We discussed removing Rekor altogether, so I'm not certain we need this.

cc @redhat-appstudio/enterprise-contract

Reply:
ah, you are setting it -> https://github.com/redhat-appstudio/infra-deployments/blob/8b04b44311fa3594af5c18f8ab3362853f022bdf/hack/preview.sh#L140 but the change is reverted by argocd...

So basically these three lines can be removed. Waiting on reaction of rest of your team.

Reply:
If you run `./hack/bootstrap.sh preview` today, it does set the transparency.url correctly and the in-cluster instance of Rekor is used. So it looks like the change is not reverted by argocd. I _think_ argocd ignores fields in a configmap that are not defined in git. Since [chains-config.yaml](https://github.com/redhat-appstudio/infra-deployments/blob/main/components/build/tekton-chains/chains-config.yaml) doesn't set the transparency.url, argocd ignores its value altogether.

The staging instance of appstudio does not have the transparency.url set because it was not bootstrapped with the preview command.

Reply:
When that was written in `preview.sh`, the only way it seemed to get the changes of the rekor url to persist were to make the changes in the preview branch and then commit them. This would then allow argo-cd to pick them up and implement them. Upon going back to upstream, the commits in the preview branch would be disregarded and the `transparency.url` field would be removed from the chains-config` secret in the cluster.

Reply:
Ok, so what is the result, should we point to in-cluster rekor for development and testing environments (using preview.sh)? I suppose that this is not blocking this PR since the behavior is not changed.

Reply:
Let's do what you have here in the PR. We want to use internal Rekor when running in preview mode.